### ISSUE TITLE
Convert amount to float before json_encode since 1.2.3.0

### ISSUE THREAD

**Issue by hjerichen** [heidelpay/heidelpayPHP#164]
I had an issue after upgrading from 1.2.0 to 1.2.5:

The $amount for authorization and such is now stored as float.
This is a problem with standard php configuration for 7.0 (serialize_precision: 17)

```
$amount = 2824.55;
var_dump(json_encode([$amount])); //output: "[2824.5500000000002]"
```

We are now on 7.4 but kept this configuration after several php version upgrades.
With the standard value of serialize_precision: -1 in 7.4 this isn't an issue.

But this configution of -1 got introduced in 7.1.
Therefore I suggest to keep $amount as a string or remove 7.0 compatibility.

If you keep it as float, maybe a warning when serialize_precision is not -1 is a good idea as well.

https://wiki.php.net/rfc/precise_float_value

**Comment by sixer1182** [comment]
Hi @hjerichen 
thanks for the hint.
We actually inform about this in the documentation: https://docs.heidelpay.com/docs/installation
However I was not aware about the incompatibility with PHP7.0.
We will look into this.

Thanks again
Simon

**Comment by hjerichen** [comment]
Oh right. I missed this in your documentation. I don't know how, because it's quite visible.

Well, at least I brought the 7.0 incompatibility to your attention. :-)

Heiko

**Comment by sixer1182** [comment]
Hi @hjerichen,
we are planning the removal of PHP7.0 compatibility in the next release.
Thank you for helping us making the SDK more reliable.
We are always greatful for such input.
So if you come across anything do not hesitate to tell us.
Have a nice weekend
Simon

**Comment by sixer1182** [comment]
FYI
This is addressed in PR #168

**Comment by sixer1182** [comment]
The change has been released in Version [1.2.7.1](https://github.com/heidelpay/heidelpayPHP/releases/tag/1.2.7.1)
